### ISSUE TITLE
panic when passing function to t.Run()

### ISSUE THREAD

**Issue by sbueringer** [kulti/thelper#31]
Hey folks,

I just upgraded to golangci-lint v1.44.0 which contains thelper v0.5.0. This version of thelper 
panics when running it against our repository (ClusterAPI). The previous version (v0.4.0) didn't.

The code block in ClusterAPI that triggers the panic seems to be:
```go
t.Run("for Cluster", customDefaultValidateTest(ctx, c, webhook))
```
https://github.com/kubernetes-sigs/cluster-api/blob/82ab810646f11ec7c96a1543697264deaa3884ec/internal/webhooks/cluster_test.go#L352

thelper panics because `f.Obj` is nil here: 
```go
		funObjDecl, ok := f.Obj.Decl.(*ast.FuncDecl)
```
https://github.com/kulti/thelper/blob/master/pkg/analyzer/analyzer.go#L415

# How to reproduce

Clone cluster-api (https://github.com/kubernetes-sigs/cluster-api). Checkout commit `4c4b47d`

cd into the repo and run:
```bash
/thelper ./internal/webhooks/...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x129572d]

goroutine 6016 [running]:
github.com/kulti/thelper/pkg/analyzer.unwrapTestingFunctionBuilding(0x12c0700, {0x1373088, 0xc0014aa7c0}, {0x13729a0, 0xc002ddf9e0})
        /Users/buringerst/code/src/github.com/kulti/thelper/pkg/analyzer/analyzer.go:415 +0x6d
github.com/kulti/thelper/pkg/analyzer.extractSubtestExp(0xc002f03520, 0xc0014aa800, {0x137ddf8, 0xc00067dbd0}, {0x13729a0, 0xc002ddf9e0})
        /Users/buringerst/code/src/github.com/kulti/thelper/pkg/analyzer/analyzer.go:394 +0xe5
github.com/kulti/thelper/pkg/analyzer.thelper.run.func1({0x1372158, 0xc0014aa800})
        /Users/buringerst/code/src/github.com/kulti/thelper/pkg/analyzer/analyzer.go:147 +0xb5
golang.org/x/tools/go/ast/inspector.(*Inspector).Preorder(0xc001593878, {0xc00261b650, 0x1520700, 0x1}, 0xc00261b950)
        /Users/buringerst/code/pkg/mod/golang.org/x/tools@v0.0.0-20201114224030-61ea331ec02b/go/ast/inspector/inspector.go:77 +0x86
github.com/kulti/thelper/pkg/analyzer.thelper.run({0x12cf480}, 0xc002f03520)
        /Users/buringerst/code/src/github.com/kulti/thelper/pkg/analyzer/analyzer.go:133 +0x40b
golang.org/x/tools/go/analysis/internal/checker.(*action).execOnce(0xc001252000)
        /Users/buringerst/code/pkg/mod/golang.org/x/tools@v0.0.0-20201114224030-61ea331ec02b/go/analysis/internal/checker/checker.go:691 +0xa1e
sync.(*Once).doSlow(0x125ec29, 0xc00014c1c0)
        /usr/local/Cellar/go/1.17.2/libexec/src/sync/once.go:68 +0xd2
sync.(*Once).Do(...)
        /usr/local/Cellar/go/1.17.2/libexec/src/sync/once.go:59
golang.org/x/tools/go/analysis/internal/checker.(*action).exec(0xc001592fa8)
        /Users/buringerst/code/pkg/mod/golang.org/x/tools@v0.0.0-20201114224030-61ea331ec02b/go/analysis/internal/checker/checker.go:579 +0x3d
golang.org/x/tools/go/analysis/internal/checker.execAll.func1(0x0)
        /Users/buringerst/code/pkg/mod/golang.org/x/tools@v0.0.0-20201114224030-61ea331ec02b/go/analysis/internal/checker/checker.go:567 +0x25
created by golang.org/x/tools/go/analysis/internal/checker.execAll
        /Users/buringerst/code/pkg/mod/golang.org/x/tools@v0.0.0-20201114224030-61ea331ec02b/go/analysis/internal/checker/checker.go:573 +0x168
```

# Workaround

When I split:
```go
t.Run("for Cluster", customDefaultValidateTest(ctx, c, webhook))
```
into two statements:
```go
testFunc := customDefaultValidateTest(ctx, c, webhook)
t.Run("for Cluster", testFunc)
```
everything works as expected.

Related ClusterAPI PR to upgrade to golangci-lint v1.44 / thelper v0.5.0 (https://github.com/kubernetes-sigs/cluster-api/pull/6014)

**Comment by kulti** [comment]
Thank you for the great descriptive issue report! It have been fixed in v0.5.1.

**Comment by sbueringer** [comment]
@kulti Thx very much for the quick fix! :)
